rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1828,6 +1828,451 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
             and packageRef count = openTradeNoExecutionDetails count
             and packageRef distinct count = 1
 
+
+func Create_PartialDeliveryPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
+	inputs:
+		tradeState TradeState (1..1) <"The original trade to be closed.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+		terminationDate AdjustableOrRelativeDate (1..1) <"The new termination date.">
+		deliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade.">
+		undeliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade.">
+	output: 
+		instruction PrimitiveInstruction (1..1)
+
+	condition ContractualProduct: <"This function applies to contractual products only.">
+		tradeState -> trade -> tradableProduct -> product -> contractualProduct exists
+
+
+	add instruction -> split -> breakdown: <"Sets the second part of the split to be a new contract with a new asset payout.">
+		        [
+                    PrimitiveInstruction {
+                        quantityChange: QuantityChangeInstruction {
+                                change: undeliveredPriceQuantity,
+                                direction: QuantityChangeDirectionEnum -> Replace,
+                                lotIdentifier: empty
+                            },
+                        termsChange: Create_PartialDeliveryInstructionTermsChange(
+                                    tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+                                    effectiveDate, terminationDate
+                                ),
+                        ...
+                    }
+                ]
+	
+func Create_PartialDeliveryInstructionTermsChange:
```

Comment:
This function does exactly the same as the existing `Create_RollTermChangeInstruction`. Consider renaming the latter more generically as `Create_EffectiveOrTerminationDateTermChangeInstruction`, and use accordingly across multiple events. Remove `Create_PartialDeliveryInstructionTermsChange`.

Reply:
fixed